openedx/features/badging/tests/test_helpers.py

```diff
@@ -0,0 +1,215 @@
+        CourseEnrollmentFactory(user=self.user, course_id=self.course1.id, is_active=True)
+        CourseEnrollmentFactory(user=self.user, course_id=course2.id, is_active=True)
+        CourseEnrollmentFactory(user=self.user, course_id=course3.id, is_active=False)
+
+        courses = [
+            (self.course1.id, self.course1.display_name),
+            (course2.id, course2.display_name)
+        ]
+        trophycase_dict = badge_helpers.populate_trophycase(self.user, courses, earned_badges=mock.ANY)
+
+        expected_return_value = dict([
+            (u'test1/123/1', {'display_name': u'ABC', 'badges': 'mocked badges'}),
+            (u'test2/123/1', {'display_name': u'XYZ', 'badges': 'mocked badges'})
+        ])
+
+        self.assertEqual(expected_return_value, trophycase_dict)
+
+        # assert order of courses by display name
+        course_detail = trophycase_dict.values()
+        self.assertEqual(expected_return_value['test1/123/1'], course_detail[0])
+        self.assertEqual(expected_return_value['test2/123/1'], course_detail[1])
+
+    @mock.patch('openedx.features.badging.helpers.Badge.objects.all')
+    def test_course_badges_with_empty_badge_queryset(self, mock_badge_objects_all):
+        message = 'Raising fake exception, if badge_queryset is none'
+        mock_badge_objects_all.side_effect = Exception(message)
+
+        with self.assertRaises(Exception) as error:
+            badge_helpers.get_course_badges(user=self.user, course_id=mock.ANY, earned_badges=mock.ANY,
+                                            badge_queryset=Badge.objects.none())
+
+        self.assertEqual(str(error.exception), message)
+
+    @mock.patch('openedx.features.badging.helpers.add_badge_earned_date')
+    @mock.patch('openedx.features.badging.helpers.filter_earned_badge_by_joined_team')
+    @mock.patch('openedx.features.badging.helpers.is_teams_feature_enabled')
+    @mock.patch('openedx.features.badging.helpers.get_course_by_id')
+    def test_course_badges_successfully(self, mock_get_course_by_id,
+                                        mock_is_teams_feature_enabled,
+                                        mock_filter_earned_badge_by_joined_team,
+                                        mock_add_badge_earned_date):
+        mock_get_course_by_id.return_value = mock.ANY
+        mock_is_teams_feature_enabled.return_value = False
+        mock_filter_earned_badge_by_joined_team.return_value = False, list()
+
+        badge1 = BadgeFactory(type=TEAM_PLAYER[TEAM_PLAYER_ENTRY_INDEX], threshold=2)
+        badge2 = BadgeFactory(type=self.type_conversationalist, threshold=2)
+        badge3 = BadgeFactory(type=self.type_conversationalist, threshold=5)
+
+        badges = badge_helpers.get_course_badges(self.user, self.course1.id, earned_badges=mock.ANY)
+
+        expected_result = {
+            'badges': {
+                'conversationalist': [{
+                    'description': None, 'image': badge2.image, 'threshold': 2L,
```

Comment:
what is this L?

Reply:
**L** is long integer literal. It makes integer [long integer](https://docs.python.org/2.0/ref/integers.html). Though it is not necessary to make threshold long integer here, integer is just fine but it doesn't break anything